File: dependency-mapper/dependency-mapper/src/main/java/com/example/mapper/adapter/RouterLogAdapter.java
package com.example.mapper.adapter;

import com.example.mapper.adapter.builders.RouterLogBuilder;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

public class RouterLogAdapter implements DataSourceAdapter<RouterLog> {
    private final Path logFile;
    private final RouterLogBuilder builder = new RouterLogBuilder();

    public RouterLogAdapter(Path logFile) {
        this.logFile = logFile;
    }

    @Override
    public RouterLog fetch() throws IOException {
        if (logFile == null || !Files.exists(logFile)) {
            return null;
        }
        String raw = Files.readString(logFile);
        return builder.parse(raw).build();
    }
}

File: service/adapter/RouterLogAdapter.js
const DataSourceAdapter = require('./DataSourceAdapter');
const RouterLogBuilder = require('./builders/RouterLogBuilder');
const fs = require('fs');

class RouterLogAdapter extends DataSourceAdapter {
  constructor(options = {}) {
    super(new RouterLogBuilder());
    this.logFile = options.logFile || '';
  }

  fetch() {
    if (!this.logFile || !fs.existsSync(this.logFile)) {
      return null;
    }
    const raw = fs.readFileSync(this.logFile, 'utf8');
    return this.parse(raw);
  }
}

module.exports = RouterLogAdapter;
